lib/document-collection.js

```diff
@@ -0,0 +1,64 @@
+const files = require('./files');
+const shortid = require('shortid');
+
+/**
+ * Need to add documentation here!
+ */
+
+
+class DocumentCollection {
+  constructor(folder) {
+    this.folder = folder;
+  }
+
+  
+  /**
+   * Save the object as a file in the document collection or throw an error
+   * @param {*} input
+   * @throws {CastError} throws if input does not conform to type
+   * @returns {Number}
+   * @function castNumber
+   */
+  save(object) {
+    const id = shortid.generate();
+    object.id = id;
+    const content = JSON.stringify(object);
+    const filename = `${this.folder}/${id}.json`;
+
+    return files.writeFile(filename, content)
+      .then(
+        () => object,
+        err => Promise.reject(err)
+      );
+  }
+
+  get(id) {
+    const filename = `${this.folder}/${id}.json`;
+    return files.readFile(filename)
+      .then(
+        content => JSON.parse(content),
+        err => Promise.reject(err)
+      );
+  }
+
+
+
+  getAll() {
+    return files.readdir(this.folder)
+      .then(
+        arr => {
+          return Promise.all(arr.map(filename => {
+            const id = filename.slice(0, filename.length - '.json'.length);
```

Comment:
nice work with this! you can alternatively use the built in "path" module to more easily manipulate file names and pathing rather than slicing, but this totally works.